### PR TITLE
BUGFIX: Make ESLint work with TypeScript again and fix issues

### PR DESCRIPTION
**Pull request by Sebobo**
Adjusted the eslintrc and ran autofix

### CODE REVIEW

**Comment by Sebobo** [comment]
@mhsdesign I dare you to revert this!

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        "no-use-before-define": "off",
+        "default-case": "off",
+        "no-mixed-operators": "off",
+        "no-negated-condition": "off",
+        "complexity": "off",
+    },
+    overrides: [
+        {
+            files: [
+                "**/*.ts",
```
Comment (mhsdesign) [inline .eslintrc.js:32]: i think we need to change this in the esbuild branch to also include js, as we use the typescript decorators there ... (and jsx)

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    env: {
+        node: true,
+    },
+    plugins: [
+        "@typescript-eslint",
+    ],
+    rules: {
+        // Fix for incorrect unused var detection
+        "no-unused-vars": "off",
```
Comment (mhsdesign) [inline .eslintrc.js:19]: :P

**Comment by mhsdesign** [comment]
> @mhsdesign I dare you to revert this!

lets see :P

**Review by markusguenther — APPROVED** [review]
Looks good to me, and it seems that everything is still working. Thanks for the work 💚

**Comment by mhsdesign** [comment]
There will be a merge conflict with #3200 

https://github.com/neos/neos-ui/compare/bugfix/ts-lint...feature/esbuild?expand=1

but fine by reading ...

**Pull request merged**